app/models/invoice.rb

```diff
@@ -287,6 +288,18 @@ def should_assign_sequential_id?
     status_changed_to_finalized?
   end
 
+  def trigger_invoice_sync
```

Comment:
Not sure we should trigger business logic from the model. What do you think of having this called from the services?

Reply:
Here in the [followup PR](https://github.com/getlago/lago-api/pull/2013/files) you can see that those methods should be called after save, so in this case I think it belongs in the model.

Reply:
That's my point :). Having callbacks in model to sync with external services is usually not a good practice.

I would suggest to add the logic in the service dealing with the create/update of the invoice, or in last resorts into an observer

Reply:
Happy to discuss it if you want :)

Reply:
I changed it so the model checks if it's possible to sync, then all the services that handle saving invoices will need to call `should_sync_invoice?` and `should_sync_sales_order?` and enquee the jobs.